Radar docs **DRAFT** (#3803)
redis/docs: content/operate/radar/install.md

---
title: Install Radar
alwaysopen: false
categories:
- docs
- operate
- radar
description: Install Radar on RHEL, Kubernetes, or Docker Compose, then make it reachable from your network.
linkTitle: Install
weight: 10
---

> Radar runs as two services backed by a PostgreSQL database that you provide:

Reviewer: we either set the db for you, or allow you to use your own

- **API server** serves the REST API and the web UI.
- **Worker** collects cluster state on a schedule.

Both services read the same database and the same encryption key, and you supply both whichever install method you choose.

> {{< warning >}}

Author: Fixed — converted to a note and reworded into two shorter sentences.

> Plan for remote access before you start. Every install method leaves network access up to you. The RPM listens only on loopback, so **an install that succeeds is still unreachable from any other machine** until you put a proxy or ingress in front of it. See [Provide remote access](#provide-remote-access).

Reviewer: we support two installation types, RPM and K8 (which you acknowledge below), but the comment here is for a specific one

Author: Warning is now RPM-specific. One question: I kept the Docker Compose sections, since DEPLOYMENT.md builds a versioned Compose bundle as an on-prem artifact alongside RPM and K8s. Is Compose customer-supported, or should I drop it?

{{< /warning >}}

## Choose an install method

| Method | Use it when | Containers required |
|---|---|---|
| [RPM on RHEL](#install-on-rhel-with-the-rpm) | You run RHEL 9 and cannot or do not want to run containers. | No |
| [Kubernetes with Helm](#install-on-kubernetes-with-helm) | You already run Kubernetes or OpenShift. | Yes |

> | [Docker Compose](#install-with-docker-compose) | You want a single host and already run Docker. | Yes |

Reviewer: lets remove docker for now, it was used for preview, I don't think customers would use it for live setup

All three are supported and built from the same release. You can install any of them on a host with no internet access. See [Install on an air-gapped host](#install-on-an-air-gapped-host).

To get the artifacts, contact your Redis account team.

## Before you start

### Supported platforms

> | Method | Requirement |

Author: Addressed — the separate Supported platforms table is gone; the requirement now lives in the install-method table above.

|---|---|
| RPM | RHEL 9 on x86_64 |
| Helm | Kubernetes 1.23 or later, and Helm 3.x. Validated on OpenShift 4.x |
| Docker Compose | A Docker engine with the `docker compose` plugin |

### PostgreSQL

> Radar stores all of its state in PostgreSQL. You own that database. Radar does not initialize, start, tune, back up, or remove it.

Reviewer: Radar does optionally  initialize, start

Use an external, managed PostgreSQL for production. The Helm chart and the Compose bundle can each start a PostgreSQL container for evaluation, but neither is hardened for production use.

The connection string needs privileges for both normal runtime work and schema migration, including `CREATEROLE`. On startup, the API server creates the roles it needs before it begins serving traffic.

Use `sslmode=require` or stricter to encrypt the connection. Radar passes your connection string through unchanged.

### The credential encryption key

> Radar encrypts the cluster credentials you give it. Each tenant gets its own data key, and all of those keys are wrapped by one key-encryption key (KEK) that you supply: **32 raw bytes**, not base64 and not hex.

Reviewer: tenant is mostly a cloud thing, in local deployment you have a single default tenant

> Radar encrypts the cluster credentials you give it. Each tenant gets its own data key, and all of those keys are wrapped by one key-encryption key (KEK) that you supply: **32 raw bytes**, not base64 and not hex.

Author: Addressed — reworded to "The key must be 32 raw bytes, not base64 or hex."

> {{< warning >}}

Author: Addressed — expanded with the fails-closed explanation.

Back up the KEK alongside the database and store the backup separately. Either one alone is useless. Credentials sealed with the old key cannot be decrypted if the API server and the worker read different keys, or if a restored database is paired with the wrong key. Radar fails closed rather than silently losing them.
{{< /warning >}}

### FIPS mode

> If you need FIPS 140-3 validated cryptography, decide before you install: it is a separate build of Radar, not a setting you turn on afterwards. Contact your Redis account team for the FIPS variant.

Author: Addressed — reworded.

Set `MCM_REQUIRE_FIPS=true` to make FIPS mandatory. Radar then refuses to start unless FIPS is actually active, and it checks before it touches the database or opens a port. A misconfigured deployment fails immediately rather than running with cryptography you did not approve.

Each service logs its FIPS state once at startup, so you can confirm what is running:

> 

Author: Fixed — clarified as "the MCM_REQUIRE_FIPS=true environment variable."

```text
fips state service=mcm-api category=startup enabled=true required=true
```

> `enabled` is the cryptography actually in effect. `required` is what you asked for. Both should read `true`.

Author: Addressed — added "In that line," to connect it to the example above.

### Sizing

The worker is the usual bottleneck. It processes jobs concurrently, and every worker shares one durable queue in PostgreSQL, so adding workers adds throughput. Raise worker concurrency and add replicas as your fleet grows.

On Kubernetes, these are the production starting points the Redis-hosted deployment uses:

| Component | CPU | Memory |
|---|---|---|
| API server | 1 | 1Gi |
| Worker | 1 | 1Gi |
| Migration job | 500m | 256Mi |

Run at least two API and two worker replicas for production, and set CPU requests before you turn on autoscaling. Kubernetes computes utilization from requests, so an autoscaler without them cannot make useful decisions.

<!-- TODO(DOC-6911): the RED-197466 GA bar is 300 clusters, but the source repo has no validated sizing for that scale (docs/DEPLOYMENT.md:266 only gives a t3.medium staging example). Ask Guy for tested numbers, then add a fleet-size sizing table. Do not extrapolate. -->

### Package and service names

> Radar's packages, services, and paths use an `mcm` prefix. The RPM is named `mcm`, its services are `mcm-api` and `mcm-worker`, its configuration lives in `/etc/mcm/`, and the container images are `mcm-app`, `mcm-worker`, and `mcm-migrate`. All of these are Redis Radar.

Reviewer 2: ```suggestion
Radar's packages, services, and paths use an `mcm` prefix. The RPM is named `mcm`, its services are `mcm-api` and `mcm-worker`, its configuration lives in `/etc/mcm/`, and the container images are `mcm-app`, `mcm-worker`, and `mcm-migrate`.
```

Author: Addressed — dropped that sentence.

## Install on RHEL with the RPM

The RPM installs native binaries and needs no container runtime. It also installs the `mcmctl` diagnostics command.

### 1. Install the package

```bash
sha256sum -c SHA256SUMS
sudo dnf install -y ./mcm-<version>-<release>.x86_64.rpm
```

The package depends on RHEL's `postgresql-server`, so `dnf` installs PostgreSQL software if it is absent. It does not create or start a database.

> The package deliberately installs its services stopped and not enabled. Confirm that before you configure anything:

Reviewer 2: There are also a few command line snippets below that treat the command line as part of the text rather than introducing them with "use the following commmand line", etc.
```suggestion
The package deliberately installs its services without starting or enabling them. Use the following command lines to confirm that they are still inactive before you configure anything:
```

Author: Addressed — reworded to "Use the following commands to confirm they are still inactive."

```bash
systemctl is-active mcm-api.service || true
systemctl is-active mcm-worker.service || true
```

### 2. Configure the package

Edit `/etc/mcm/mcm.env` and set the two required values:

```bash
sudoedit /etc/mcm/mcm.env
```

| Setting | Description |
|---|---|
| `DATABASE_URL` | Connection string for your PostgreSQL database, with runtime and migration privileges. |
| `CREDENTIAL_ENCRYPTION_KEY` | The credential encryption key. |

Radar refuses to start while the placeholder values are still in place. The file is owned by `root:mcm`, redacted from logs and diagnostics, and kept across upgrades and removal. Include it in your backup plan.

> Useful defaults you may want to change:

Reviewer 2: ```suggestion
There are a few useful defaults that you can change if necessary:
```

Author: Addressed — already reworded to "You may want to change these defaults:", which covers this.

### 3. Start the services

Check the configuration first:

```bash
sudo mcmctl doctor
```

Before the first start, `mcmctl doctor` reports that schema migration has not run yet. That is expected; the configuration and database connectivity checks should still pass.

```bash
sudo systemctl start mcm-api.service
sudo systemctl start mcm-worker.service
```

The API server applies the database migrations as it starts. Verify it is serving:

```bash
curl -fsS http://127.0.0.1:8080/healthz/ready
sudo mcmctl doctor
```

### 4. Enable the services

Enable the services only after the health checks pass, so a reboot cannot start a half-configured deployment:

```bash
sudo systemctl enable mcm-api.service
sudo systemctl enable mcm-worker.service
```

### 5. Create the first administrator

Radar does not ship a default account or a default password. Open the UI once the API server is healthy and complete the one-time first-administrator flow. It is available only while the database has no users; after that, it closes and normal sign-in applies.

> At this point Radar is reachable only from the host itself. Continue to [Provide remote access](#provide-remote-access).

Reviewer 2: Since the "Provide remote access" section is split into RPM and K8s subsections, it might be better to include that information as another subsection here rather than saying "Continue to `Provide remote access`" (saves a separate section and provides the info right where it is relevant).

Author: Fixed — folded remote access into each install method as its own step/subsection (RPM gets step 6, Kubernetes gets a subsection after OpenShift) and removed the standalone section.

## Install on Kubernetes with Helm

A production install has four parts you supply: the PostgreSQL connection, the credential encryption key, image pull access, and an external access path.

### 1. Create the secrets

Store the database connection string in a secret:

```bash
kubectl create secret generic radar-db \
  --namespace radar \
  --from-literal=DATABASE_URL='postgres://radar:secret@postgres.example.com:5432/radar?sslmode=require'
```

Generate the credential encryption key as a file and load it with `--from-file`:

```bash
head -c 32 /dev/urandom > kek.bin

kubectl create secret generic radar-credentials \
  --namespace radar \
  --from-file=CREDENTIAL_KEK=./kek.bin

shred -u kek.bin
```

{{< note >}}

> Write the key to a file rather than using `--from-literal="$(head -c 32 /dev/urandom)"`. Command substitution truncates at null bytes, so the key would not be 32 bytes.

Reviewer 2: ```suggestion
Write the key to a file rather than using `--from-literal="$(head -c 32 /dev/urandom)"`. If the key contains bytes with a value of zero then the command substitution will treat them as null characters and truncate the key at the first one it finds.
```

Author: Addressed — reworded to "a zero byte... truncates it there, so the key would be shorter than 32 bytes."

{{< /note >}}

> The secret must contain the `CREDENTIAL_KEK` key. If it does not, the pods stay in `ContainerCreating`. That failure is deliberate rather than silent.

Reviewer 2: Not very clear.

Author: Addressed — reworded with the reason included: "rather than starting with no encryption key."

```bash
sudo systemctl restart mcm-api.service
sudo mcmctl doctor
```

`mcmctl doctor` checks runtime health through the configured address. If it reports a runtime-health failure after you change the listen address, confirm the service bound to the interface you expected and that the proxy forwards to the same address.

{{< warning >}}
Do not expose Radar directly on a public interface. Terminate TLS and apply access controls at the edge.
{{< /warning >}}

### Kubernetes

The API server and UI are served on port 80 of an in-cluster service. Expose it with an ingress, an OpenShift route, or a `LoadBalancer` service, and terminate TLS there:

```yaml
ingress:
  enabled: true
  className: nginx
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt-prod
  hosts:
    - host: radar.example.com
      paths:
        - path: /
          pathType: Prefix
  tls:
    - secretName: radar-tls
      hosts:
        - radar.example.com
```

Radar marks the browser session cookie as secure by default, so serve Radar over HTTPS. Over plain HTTP the browser rejects the cookie and sign-in fails.

## Install on an air-gapped host

> Air-gapped installation is not a separate method. It is the same three methods, with the artifacts carried in by hand instead of downloaded.

Reviewer 2: Reads awkwardly.

Author: Addressed — reworded into two sentences.

Transfer the release artifacts to the target host or to an offline repository it can reach, then verify them:

```bash
sha256sum -c SHA256SUMS
```

| Method | What to transfer | How it installs |
|---|---|---|
| RPM | The `.rpm`, `SHA256SUMS`, and the dependency closure, including `postgresql-server` if the host has no offline PostgreSQL | `dnf install` from the local file |
| Helm | `images.tar.gz`, the packaged chart, and the bundled values file | `docker load` the images onto the nodes, then install the chart |
| Docker Compose | `images.tar.gz` and the Compose files | `docker load`, then `docker compose up` |

The RPM is the one path that needs no container tooling at all. It installs, starts, and upgrades with no internet access, no online package repositories, and no Docker, Podman, or containerd on the host.

Your PostgreSQL database and the clusters you plan to monitor still need to be reachable from the Radar host over the network.

## Next steps

> Radar is installed but has nothing to show yet. It does not discover clusters on its own. Continue to [Connect clusters]({{< relref "/operate/radar/connect" >}}) to add your first cluster.

Reviewer 2: ```suggestion
At this point, Radar is installed but has nothing to show yet because it does not discover clusters on its own. Continue to [Connect clusters]({{< relref "/operate/radar/connect" >}}) to add your first cluster.
```

Author: Addressed — applied (dropped the leading "At this point,").
